features/ifttt/src/test/java/org/opennms/features/ifttt/IfTttDaemonTest.java

```diff
@@ -0,0 +1,275 @@
+                this.oc = Integer.valueOf(variableNameExpansion.replace("%oc%"));
+                this.nc = Integer.valueOf(variableNameExpansion.replace("%nc%"));
+                os = variableNameExpansion.replace("%os%");
+                ns = variableNameExpansion.replace("%ns%");
+            }
+        }
+
+        public ResultEntry(final String event, final String os, final Integer oc, final String ns, final Integer nc) {
+            this.event = event;
+            this.os = os;
+            this.ns = ns;
+            this.oc = oc;
+            this.nc = nc;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("%s : %s/%d -> %s/%d", event, os, oc, ns, nc);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            ResultEntry that = (ResultEntry) o;
+
+            if (oc != that.oc) return false;
+            if (nc != that.nc) return false;
+            if (!os.equals(that.os)) return false;
+            if (!ns.equals(that.ns)) return false;
+            return event.equals(that.event);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = os.hashCode();
+            result = 31 * result + ns.hashCode();
+            result = 31 * result + oc;
+            result = 31 * result + nc;
+            result = 31 * result + event.hashCode();
+            return result;
+        }
+    }
+
+    private Map<Integer, OnmsNode> nodeMap;
+    private Map<Integer, OnmsAlarm> alarmMap;
+    private Map<String, OnmsCategory> categoryMap;
+
+    private void addCategory(String category) {
+        categoryMap.put(category, new OnmsCategory(category));
+    }
+
+    private void addNode(Integer id, String label, String... categories) {
+        final OnmsNode onmsNode = new OnmsNode();
+        onmsNode.setId(id);
+        onmsNode.setLabel(label);
+        onmsNode.setCategories(Arrays.stream(categories).map(c -> categoryMap.get(c)).collect(Collectors.toSet()));
+        nodeMap.put(id, onmsNode);
+    }
+
+    private void addAlarm(Integer id, Integer nodeId, OnmsSeverity onmsSeverity) {
+        addAlarm(id, nodeId, onmsSeverity, false);
+    }
+
+    private void addAlarm(Integer id, Integer nodeId, OnmsSeverity onmsSeverity, final boolean acknowledged) {
+        final OnmsAlarm onmsAlarm = new OnmsAlarm(id, EventConstants.NODE_LOST_SERVICE_EVENT_UEI, null, null, onmsSeverity.getId(), new Date(), new OnmsEvent()) {
+            @Override
+            public boolean isAcknowledged() {
+                return acknowledged;
+            }
+        };
+        onmsAlarm.setNode(nodeMap.get(nodeId));
+        alarmMap.put(id, onmsAlarm);
+    }
+
+    @Before
+    public void setup() {
+        MockLogAppender.setupLogging();
+
+        System.setProperty("opennms.home", "src/test/resources");
```

Comment:
Let's just enhance the Daemon to allow setting the config file directly instead of doing it via the system property.